src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs

```diff
@@ -1202,5 +1324,16 @@ public static void GenerateNupkgMetadataFile(string nupkgPath, string installPat
                     }
                 });
         }
+
+        static IEnumerable<T> CancellableYieldEnumeration<T>(IEnumerable<T> enumerable, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            foreach (T item in enumerable)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Have you tested if VS/PM UI emits fault telemetry when this exception is thrown?

Months ago we started implementing cancellation in PM UI to improve performance, but our fault telemetry wasn't ignoring it, becoming the number one fault by count, making it difficult for us to detect unhandled http timeout exceptions.

I hope that now we've fixed it (I created a PR a few weeks/months ago to fix some obvious problems), but you should be able to test by switching between the updates and browse tab quickly when PM UI has a local feed selected in the source dropdown.

Reply:
Currently this exception is swallowed silently, I don't know why. But did create follow up [issue#11622](https://github.com/nuget/home/issues/11622). This is one place I found there could be

Reply:
You misunderstand. We don't want fault telemetry when we intentionally cancel tasks.

Reply:
Sorry my mistake, so we're good here. No faults telemetry observed during my testing.